server/src/main/java/com/broadcom/lsp/cobol/service/CopybookServiceImpl.java

```diff
@@ -124,9 +124,8 @@ CopybookModel resolveSync(
                 .getConfiguration(COPYBOOK_RESOLVE.label, cobolFileName, copybookName)
                 .get());
     if (uri.isEmpty()) {
-      if (copybookProcessingMode.download) {
-        copybooksForDownloading
-            .computeIfAbsent(cobolFileName, s -> ConcurrentHashMap.newKeySet())
+      if (copybookProcessingMode.download && cobolFileName != null) {
```

Comment:
Anyway better to check `cobolFileName` for `null`.
`copybooksForDownloading.computeIfAbsent` if always not null, so it is false-positive issue in that case

Reply:
I think that the issue is that `computeIfAbsent` can return null if our mapping function returns null. But maybe your assumption right.

Reply:
Maybe better to check this set as well :-)

Reply:
Nope :-( Have a warning from Idea that set is always not null:
````
        Set<String> copybooks = copybooksForDownloading.computeIfAbsent(cobolFileName, s -> ConcurrentHashMap.newKeySet());
        Optional.ofNullable(copybooks).ifPresent(c -> c.add(copybookName));

````

Reply:
I agree that it will never be a null. We will see it tomorrow.